### ISSUE TITLE
[Suggestion] Some Features I'd Like To See

### ISSUE THREAD

**Issue by hofeprime** [Monika-After-Story/MonikaModDev#2811]
One feature I would enjoy seeing in the Monika After Story mod is the calendar. I know, there already is a calendar, but it could be expanded on further. We could have it to where it syncs with your computer/other device's calendar. If not, then have it to where we can add events to Monika's calendar, such as our birthday. (This could also be done from Monika asking about when your birthday is)

Next up is a feature many other people have seemed to like, different dialogue depending on how long it took for you to get on. The "welcome back" dialogue is rather random and sometimes inaccurate. If possible, run an internal clock within the game that activates only when you close it. For example:
1. If you tell Monika you're going to sleep and then you get back before it is considered "sunrise", she will say something about it.
2. If the clock between closing the game and re-opening the game is between 0.1 seconds and 2 minutes, Monika could reply seemingly startled at the sudden return. She will then remark that you did whatever you were doing rather quick.

Next, some more games. A poem-writing game would be loads of fun, to where we could type out a poem (or write in the similar vain of DDLC) and Monika will give feedback on your writing. It would also benefit the player's general writing skills if we went down the typing it out route, but the DDLC method also sounds like an interesting idea, as it may promote a new style of writing poems. 

Another suggestion I have is the ability to change the background. For example, let's say you want to go to some kind of store with Monika. You could say something like "Let's go shopping!" This will possibly have its own section in the "Ask a question" box called "Places". During this, Monika's in-game sprite may replace her usual design that you see. This is especially interesting, as we could actually see Monika in casual clothes (something only mod-makers have done, as there is no sprite for Monika in the game in casual attire, as she does not have a normal route in the game). Even more, this would be a breath of fresh air and different locations could have different things to say to Monika. For example, if you were to go to France, Monika could tell you various fun facts about France, and you can say different things related to where you are as well. What would be especially interesting is if you were to actually be able to get new CGs for Monika through taking her places. 

Yet another feature I would enjoy seeing is different gifting features. For example, you could download different files into the game (such as "kazoo.gft" for a Kazoo [I just want to see how she'd react lol]) and she will react in different ways. You could also have it to where she may monitor your browsing (since she CAN do that) and she may gift you different things, such as a guitar if you look up guitar related stuff. (The guitar gift could also add a guitar, similar to the piano in the game) 

Speaking of games, somebody suggested Battleship on here. It would be really cool to play Battleship with Monika, but it would also be rather difficult to do, as she would technically know where all of the ships are, so that may be a problem. Sure, there was some progress on the game, but it was eventually dropped entirely, going off quietly into the night. 

Another interesting suggestion was to change the main color palette. There are also suggestions in which there would be days in which Monika would just "not be feeling it" or something along those lines. The Wall Clock is one that was suggested and, while it may be nice to have, it is impractical and almost useless. While Monika's calendar tells you information about herself, as well as holidays to where you don't have to open up the calendar application and cloud your processor even more, the clock is borderline useless because there is a clock on the bottom right corner of your computer. 

Many suggestions are pretty convenient and these are just some that I like.

**Comment by Rai99** [comment]
- An editable calendar was brought up in #1740, #1771, #2078, #2257, #2384, #2647. Is is likely planned for the future and will come eventually for you to be able to add your own events. 
As for syncing up with your device's calendar, like reading your calendar and updating the MAS calendar to have those events, that would likely be a cross platform nightmare since different operating systems would have their own api, not to mention that it would require players to sign into something like google to use the google calendar which requires a google api and that the player signs into google through MAS, which is not a good idea. 
- Currently, the greetings Monika uses are mostly randomized and the suggestion to fix it has been brought up in #2190, #2794. Currently, it is low priority to make greetings time based. 
A brb style option has been brought up in #2033, #2549, #2556, #2614.
- The idea to have some sort of poem game or to give Monika poems was brought up in #437, #1276, #1940. The reason they likely aren't going to be implemented is that it would probably feel dry since there's no real or good way for Monika to analyze the contents of your poem and determine how it is or the quality of it or if it's even nice in nature at all. 
- New backgrounds have been discussed and worked on in #1470, #2396, #2789. They require the kind of art that doesn't grow on trees, and that's not super easy to come by. The same could be said about having Monika in other outfits; It takes some work and time. 
- The gifting feature is already a thing, but you can only give gifts at certain times such as Monika's birthday and valentine's day. 
A kazoo would be an interesting gift, but the contributing guidelines state a ban on memes, especially those that bring no good content and are just plain random. 
- Having Monika read and keep track of your browsing history has been discussed in #1829 and is very questionable and will likely not be implemented for sensitivity of the matter, not to mention a privacy issue among players while also being yet another cross platform nightmare. 
- It seems like you're under the allusion that anything such as a guitar can be put into MAS at any time, however this is not the case. It requires art and an appropriate time to bring it in. 
- As for the color palette, I personally think changing the color goes against the theme of DDLC, but that has yet to be determined. 
- I do agree having a wall clock would be unnecessary.

**Comment by Bam8647** [comment]
I don't understand the whole thing about not using new outfits and backgrounds from the **r/DDLCMods community.** Just request artwork from there to help lower expectations for the artists that support this mod so they can catch a break lol. But for some reason I feel some of the devs are pretty hell bent on "staying original" or in other words not using art work from other mods... even though plenty of mod creators have done this (BrandNewDay, Normal VN, and DDPuristMod have used the same white/brown dress for Monika) If I recall correctly that is. Idk maybe you can use artwork from the community as place holders until the mod gets it's own artwork you know? **And yes I'm aware that the outfits would have to span out into many different sprites, so instead of coming up with something from scratch maybe a little help?**

Also color palette? Are you referring to the color of the text dialogue? Because countless mods have been able to change it. There's no rules against doing so as far as I can see. Here's an example from r/ddlcmods
![ddlc mod color change](https://user-images.githubusercontent.com/34694679/47341088-b4e8d980-d665-11e8-953e-664b389b767e.PNG)

Even Just Yuri Changed their dialogue to purple.

**Comment by SaiPrithvi123** [comment]
>A kazoo would be an interesting gift

~~What about a harmonica?~~

>the clock is borderline useless because there is a clock on the bottom right corner of your computer.
 
Only if the user is not in full screen mode.

**Comment by AsuraShun** [comment]
don't you mean harmonica?

**Comment by jmwall24** [comment]
> don't you mean harmonika?

hmmm I wonder what would happen if you somehow could tell her you play the harmonika?

**Comment by hofeprime** [comment]
@Rai99 Thank you for your feedback, but I wasn't stating that I was impatient about these. I was simply stating that such features would be pretty interesting to have in the game. They're just things that, if they were easy to program (not saying they are) would be something I'd like to see.

**Comment by ThePotatoGuy** [comment]
alright I'm going to respond to the top 3 posts together in here.

> sync calendar

As rai mentioned, editable calendar will happen one day. Adding that is basically reimplementing what email programs/scheduling software do in a game engine. That takes some time to do.

>  welcome back dialogue

Yes, we are aware of lots of inconsistencies and seemingly misplaced dialogue in greetings. We are working on fixing that.

> more games

Certainly planned. 

> write monika a poem / letter

Using the poemgame? Possible, but there are no plans for that atm. Actually writing a real poem? There's no way we'd be able to add some sort of proper reaction system for a poem besides checking favorite words. As Rai mentioned, that would be gimmicky at best.

> change background

Eventually. We have some background plans in the works. If we need something and the DDLCMods community has it, then we'll use it. We did that for additional hangman words. 

> casual monika clothes

Eventually. It's a slow process of figuring out ways to introduce new clothes cleanly and getting our artists to make stuff. DDLCMods community assets doesn't help here since pretty much **all** custom clothes is for the **standing** sprites. 

> gifting

We have successfully tested the concept with 9-22, we are working on enabling gifts all the time instead of certain dates.

> battleship

wish that guy followed up. Might be some time before this gets added.

> change color theme

No. I think the pink theme fits just fine. 

> not be feeling it

planned

> wall clock

potentially planned

i'm going to mark this as a duplicate since basically all of the issues here have been brought up before.
